File: JavaScript solutions/191. Number of 1 Bits.js
//approach : 1
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function (n) {
  const str = n.toString(2);
  let one = 0;
  for (let char of str) {
    if (char === "1") one++;
  }
  return one;
};

//approach : 2
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function (n) {
  return n.toString(2).replaceAll("0", "").length;
};

File: C++ solutions/191. Number of 1 Bits.cpp
//approach : 1
class Solution {
public:
    int hammingWeight(uint32_t n) {
      int one=0;
      for(int i=1;i<=32;i++){
          int a=n%2;
          if(a==1) one++;
          n/=2;
      }
      return one;
    }
};

//approach : 2
class Solution {
public:
    int hammingWeight(uint32_t n) {
        return bitset<32>(n).count();
    }
};